dot/state/epoch.go

```diff
@@ -220,8 +240,32 @@ func (s *EpochState) SetEpochData(epoch uint64, info *types.EpochData) error {
 	return s.db.Put(epochDataKey(epoch), enc)
 }
 
-// GetEpochData returns the epoch data for a given epoch
-func (s *EpochState) GetEpochData(epoch uint64) (*types.EpochData, error) {
+// GetEpochData returns the epoch data for a given epoch persisted in database
+// otherwise will try to get the data from the in-memory map using the header
+// if the header params is nil then it will search only in database
+func (s *EpochState) GetEpochData(epoch uint64, header *types.Header) (*types.EpochData, error) {
+	epochData, err := s.GetEpochDataInDatabase(epoch)
+	if err == nil && epochData != nil {
+		return epochData, nil
+	}
+
+	if err != nil && !errors.Is(err, chaindb.ErrKeyNotFound) {
+		return nil, fmt.Errorf("failed to get epoch data from database: %w", err)
+	} else if header == nil {
+		// if no header is given then skip the lookup in-memory
+		return epochData, nil
+	}
+
+	epochData, err = s.GetEpochDataForHeader(epoch, header)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get epoch data from memory: %w", err)
```

Comment:
nit should `GetEpochDataForHeader` be renamed to `GetEpochDataInMemory` so it uses the same naming format as `GetEpochDataInDatabase`?